### DISCUSSION TITLE
glsp-server-node

### DISCUSSION THREAD

**Discussion by Mansalek** [eclipse-glsp/glsp#560] [Q&A] [answered]
Hi,
I have been trying to run the example of glsp -theia-integration but with the glsp server node (https://github.com/eclipse-glsp/glsp-server-node) .
Having  already installed all the necessary dependencies and built the project as specified , I'm still not able to run  it without errors.
Here are my logs , any help would be appreciated .
![logs](https://user-images.githubusercontent.com/100411404/155698512-27470845-dc8c-4688-9475-1f11fd0d9d62.PNG)
![logs2](https://user-images.githubusercontent.com/100411404/155698516-cb7b1b26-adb0-4834-a5fa-e89855a7f45f.PNG)
![logs3](https://user-images.githubusercontent.com/100411404/155698517-9f29a6f2-9f8b-4973-9d12-01d10052e607.PNG)

**Comment by sgraband** [comment, marked as answer]
Hey @Mansalek,

does it work with the Java-based GLSP server?

This seems like a windows path issue:

> Could not load model from file: /c:/...

I think the leading `/` should not be there.

You could try to add a `.substring(1)` here: https://github.com/eclipse-glsp/glsp-theia-integration/blob/f658c13ad3e994969b2fb8120592f929d32c1510/packages/theia-integration/src/browser/diagram/glsp-diagram-widget.ts#L93. If this is indeed the problem it should work in the meantime for you and we can provide a fix as soon as possible.

**Reply by Mansalek** [reply]
> Hey @Mansalek,
> 
> does it work with the Java-based GLSP server?
> 
> This seems like a windows path issue:
> 
> > Could not load model from file: /c:/...
> 
> I think the leading `/` should not be there.
> 
> You could try to add a `.substring(1)` here: https://github.com/eclipse-glsp/glsp-theia-integration/blob/f658c13ad3e994969b2fb8120592f929d32c1510/packages/theia-integration/src/browser/diagram/glsp-diagram-widget.ts#L93. If this is indeed the problem it should work in the meantime for you and we can provide a fix as soon as possible.

Thank you , it works perfectly right now  . Seems weird but cant' complain.
Have a good day

**Reply by sgraband** [reply]
I opened an [issue](https://github.com/eclipse-glsp/glsp/issues/561), where you can track the progress of a potential fix.

**Comment by Mansalek** [comment]
reviving this thread just to tell you that this bug(the leading / ) reoccurs within the new theia example (https://github.com/eclipse-glsp/glsp-examples/tree/master/project-templates/node-json-theia)
But this time we can't use the same workaround in the glsp-diagram-widget by adding substring(1)
Any potential fix?

**Comment by tortmayr** [comment]
I have opened a [PR](https://github.com/eclipse-glsp/glsp-theia-integration/pull/116) for the Theia integration that should fix OS related URI issues. Once the PR is merged we can update the project templates.

**Reply by Mansalek** [reply]
> I have opened a [PR](https://github.com/eclipse-glsp/glsp-theia-integration/pull/116) for the Theia integration that should fix OS related URI issues. Once the PR is merged we can update the project templates.

Does that also solve the node-json-theia project template issue? because it's the same one

**Reply by planger** [reply]
Yes it should also fix the issue in the `node-json-theia` project template. However, the `node-json-theia` project template hasn't been upgraded yet to the fixed `glsp-theia-integration` build and Theia `1.25.0`. It is still [yarn.lock](https://github.com/eclipse-glsp/glsp-examples/blob/master/project-templates/node-json-theia/glsp-client/yarn.lock)ed to 1.24.0 and a `glsp-theia-integration` version before the PR @tortmayr mentioned above.

Once we upgraded the example it should work. We should be able to do that at latest next week.

**Reply by tortmayr** [reply]
@Mansalek I have opened a [PR](https://github.com/eclipse-glsp/glsp-examples/pull/127) for the project templates. Amongst other things this should solve the pathing issue under Windows
